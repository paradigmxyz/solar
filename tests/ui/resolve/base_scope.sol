contract C {
    constructor(uint x) {
    }
}

contract D is C(x) { //~ ERROR: unresolved symbol
    uint x;
}

// OK
contract E is C {
    uint constant x = 69;
    constructor() C(x) {}
}
contract F is C() { // TODO: ~ERROR: mismatched base constructor arguments
    uint constant x = 69;
    constructor() C(x) {}
}
