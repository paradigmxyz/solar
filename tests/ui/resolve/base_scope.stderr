error: unresolved symbol `x`
  --> ROOT/tests/ui/resolve/base_scope.sol:LL:CC
   |
LL | contract D is C(x) {
   |                 ^
   |

error: aborting due to 1 previous error

